### DISCUSSION TITLE
How to dump json with single quote.

### DISCUSSION THREAD

**Discussion by kapilpipaliya** [danielaparker/jsoncons#402] [General] [answered]
Thank you so much for the great library.

I want to dump json with single quote like `"{id: 1, data: 'foo'}"` and no quotes around attributes. for example: https://stackoverflow.com/questions/20841571/how-to-solve-the-dilemma-while-handle-json-in-neo4j-cypher

can you please explain how to do it?

**Comment by danielaparker** [comment, marked as answer]
You'd have to modify `include/jsonencoder.hpp` to achieve that, making changes in two classes: `basic_json_encoder` and `basic_compact_json_encoder`. 

In the `visit_key` function, remove the two occurrences of `sink_.push_back('\"')`;

And in the `visit_string` function, change the two occurrences of `sink_.push_back('\"')` to `sink_.push_back('\'')`

**Reply by kapilpipaliya** [reply]
Thank you very much. Its working perfect.

**Comment by kapilpipaliya** [comment]
when a string has ' how to replace it with \\'  ?

**Reply by danielaparker** [reply]
In json_encoder.hpp, in escape_string, add
```
case '\'':
    sink.push_back('\\');
    sink.push_back('\'');
    count += 2;
    break;
```

**Reply by kapilpipaliya** [reply]
Thank you so much.
